### PR TITLE
[FIR] Support several annotation class diagnostics

### PR DESCRIPTION
**Pull request by McCreeMainer**
Added checker for next errors:
- ANNOTATION_CLASS_MEMBER
- LOCAL_ANNOTATION_CLASS_ERROR
- MISSING_VAL_ON_ANNOTATION_PARAMETER
- VAR_ANNOTATION_PARAMETER

### CODE REVIEW

File: compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirAnnotationClassDeclarationChecker.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.jetbrains.kotlin.KtNodeTypes.*
+import org.jetbrains.kotlin.lexer.KtTokens.*
+import org.jetbrains.kotlin.psi.KtParameter
+
+object FirAnnotationClassDeclarationChecker : FirDeclarationChecker<FirDeclaration>() {
+    override fun check(declaration: FirDeclaration, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (declaration !is FirRegularClass) return
+        if (declaration.classKind != ANNOTATION_CLASS) return
+        for (it in declaration.declarations)
+            when (it.source?.elementType) {
```
Comment (mglukhikh) [inline compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirAnnotationClassDeclarationChecker.kt:31]: Here it's much easier to check `it` type:
```
when {
    it is FirConstructor && it.isPrimary -> { ... }
    it is FirRegularClass -> { ... }
    else -> { ... }
}
```

Comment (mglukhikh) [reply compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirAnnotationClassDeclarationChecker.kt:31]: (when possible, we should not go to `source` to check something, but use own FIR properties and types)

Comment (McCreeMainer) [reply compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirAnnotationClassDeclarationChecker.kt]: > (when possible, we should not go to `source` to check something, but use own FIR properties and types)
Fixed.

**Comment by mglukhikh** [comment]
Merged manually. Congratulations!

**Pull request closed without merging**